docs/decisions/0002-inter-app-apis.rst

```diff
@@ -0,0 +1,44 @@
+Inter-app APIs
+--------------
+
+Status
+======
+
+Accepted
+
+Context
+=======
+
+The edx-platform Django project is a conglomeration of different LMS and Studio features written in the structure of Django apps. Over the years, the boundaries between features have become muddled for various reasons. We now find apps intruding into the Python innards of other apps, making the intrusive apps tightly dependent on the inner behaviors of other apps.
+
+Due to this lack of clearly delimited separation of concerns, the monolith has become hard to maneuver, comprehend, and modify.
+
+Decisions
+=========
+
+#. Each Django app should clearly define a set of Python APIs that it exposes to other Django apps, which run within the LMS/Studio process.
+
+#. Each app's Python APIs should be defined/exported in a Python module called "api.py", located in the top-most directory of the app.
+
+#. The app's Python APIs should be well-named, self-consistent, and relevant to its own "domain" (without exposing technical and implementation details).
+
+#. In the spirit of dis-entanglement and independence, an app's Django models and other internal data structures should not be exposed via its Python APIs.
+
+#. Ideally, tests should use only Python APIs declared in other apps' "api.py" files. However, if an app's API is needed only for testing (and not needed as part of the app's domain API), then test-relevant Python APIs should be defined/exported in an intentional Python module called "test_api.py".
```

Comment:
Yes, that's much better.  Also, I'm changing the proposed filename to `api_for_tests.py`.  (Since `test_api.py` is used in many places the API providers to test their APIs.  😄 )

I thought I had created a reference example in grades for `api_for_tests.py`.  But I see that wasn't needed for the grades app.